apps/emqx_management/src/emqx_mgmt_api_data_backup.erl

```diff
@@ -0,0 +1,361 @@
+                    mria_membership:coordinator()
+            end
+    end.
+
+data_files(post, #{body := #{<<"filename">> := #{type := _} = File}}) ->
+    [{FileName, FileContent} | _] = maps:to_list(maps:without([type], File)),
+    case emqx_mgmt_data_backup:upload(FileName, FileContent) of
+        ok ->
+            {204};
+        {error, Reason} ->
+            {400, #{code => 'BAD_REQUEST', message => emqx_mgmt_data_backup:format_error(Reason)}}
+    end;
+data_files(get, #{query_string := PageParams}) ->
+    case emqx_mgmt_api:parse_pager_params(PageParams) of
+        false ->
+            {400, #{code => ?BAD_REQUEST, message => <<"page_limit_invalid">>}};
+        #{page := Page, limit := Limit} = Pager ->
+            {200, #{data => list_backup_files(Page, Limit), meta => Pager}}
+    end.
+
+data_file_by_name(Method, #{bindings := #{filename := Filename}, query_string := QS}) ->
+    case safe_parse_node(QS) of
+        {error, Msg} ->
+            {400, #{code => 'BAD_REQUEST', message => Msg}};
+        Node ->
+            case get_or_delete_file(Method, Filename, Node) of
+                {error, not_found} ->
+                    {404, #{
+                        code => ?NOT_FOUND, message => emqx_mgmt_data_backup:format_error(not_found)
+                    }};
+                Other ->
+                    response(Other)
+            end
+    end.
+
+%%------------------------------------------------------------------------------
+%% Internal functions
+%%------------------------------------------------------------------------------
+
+get_or_delete_file(get, Filename, Node) ->
+    emqx_mgmt_data_backup_proto_v1:read_file(Node, Filename, infinity);
+get_or_delete_file(delete, Filename, Node) ->
+    emqx_mgmt_data_backup_proto_v1:delete_file(Node, Filename, infinity).
+
+safe_parse_node(#{<<"node">> := NodeBin}) ->
+    NodesBin = [erlang:atom_to_binary(N, utf8) || N <- emqx:running_nodes()],
+    case lists:member(NodeBin, NodesBin) of
+        true -> erlang:binary_to_atom(NodeBin, utf8);
+        false -> {error, io_lib:format("Unknown node: ~s", [NodeBin])}
+    end;
+safe_parse_node(_) ->
+    node().
+
+response({ok, #{db_errors := DbErrs, config_errors := ConfErrs}}) ->
+    case DbErrs =:= #{} andalso ConfErrs =:= #{} of
+        true ->
+            {204};
+        false ->
+            DbErrs1 = emqx_mgmt_data_backup:format_db_errors(DbErrs),
+            ConfErrs1 = emqx_mgmt_data_backup:format_conf_errors(ConfErrs),
+            Msg = unicode:characters_to_binary(io_lib:format("~s", [DbErrs1 ++ ConfErrs1])),
+            {400, #{code => ?BAD_REQUEST, message => Msg}}
+    end;
+response({ok, Res}) ->
+    {200, Res};
+response(ok) ->
+    {204};
+response({error, Reason}) ->
+    {400, #{code => ?BAD_REQUEST, message => emqx_mgmt_data_backup:format_error(Reason)}}.
+
+list_backup_files(Page, Limit) ->
+    Start = Page * Limit - Limit + 1,
+    lists:sublist(list_backup_files(), Start, Limit).
```

Comment:
Naive paging, but I think the number of stored backup files is not expected to be very large...
Additionally, this simply lists `*.tar.gz` files in `<emqx data dir>/backup`. Adding some Mnesia DB layer for backups is possible but it would be hard to keep it in sync with FS state, especially taking into account the fact that users can directly manage `backup` dir without the dashboard..